### PR TITLE
Reverting segmentation data to Unicode 15.0

### PR DESCRIPTION
**Pull request by robertbastian**
Our unversioned rules are not compatible with 15.1 properties data.

### CODE REVIEW

File: provider/datagen/src/transform/segmenter/mod.rs [hunk 1]
```diff
@@ -81,561 +81,542 @@ struct SegmenterRuleTable {
...
-            right_index: usize,
-            break_state: i8,
-        ) {
-            let index = left_index * property_length + right_index;
-            if break_state_table[index] == UNKNOWN_RULE
-                || break_state_table[index] == NOT_MATCH_RULE
-            {
-                break_state_table[index] = break_state;
-            }
+fn generate_rule_break_data(
```
Comment (Manishearth) [inline provider/datagen/src/transform/segmenter/mod.rs:84]: nit: document that this is currently always called with a different provider that uses hardcoded data

Comment (robertbastian) [reply provider/datagen/src/transform/segmenter/mod.rs:84]: Is this blocking? I think that's fairly obvious from the single call site.

Comment (Manishearth) [reply provider/datagen/src/transform/segmenter/mod.rs:84]: I had to dig a bit. Not blocking

**Review by Manishearth — APPROVED** [review]

File: provider/datagen/src/provider.rs [hunk 2]
```diff
@@ -279,6 +279,70 @@ impl DatagenProvider {
     ) -> Result<impl IntoIterator<Item = icu_locid::LanguageIdentifier>, DataError> {
         self.cldr()?.locales(levels)
     }
+
+    pub(crate) fn new_hardcoded_segmenter_data() -> Self {
+        Self {
+            source: SourceData {
+                trie_type: TrieType::Small,
+                icuexport_paths: Some(Arc::new(SerdeCache::new(AbstractFs::Memory(
```
Comment (sffc) [inline provider/datagen/src/provider.rs:287]: Thought: Good design for SerdeCache that this is so easy to do

**Review by sffc — COMMENTED** [review]
Nice work! Thanks for doing this

**Pull request merged**